pkg/piecestore/psserver/agreementsender/agreementsender.go

```diff
@@ -88,22 +94,21 @@ func (as *AgreementSender) SendAgreementsToSatellite(ctx context.Context, satID
 	//todo:  stop sending these one-by-one, send all at once
 	for _, agreement := range agreements {
 		rba := agreement.Agreement
+		// Send agreement to satellite
+		sat, err := client.BandwidthAgreements(ctx, &rba)
 		if err != nil {
-			as.log.Warn("Agreementsender failed to deserialize agreement : will delete", zap.Error(err))
-		} else {
-			// Send agreement to satellite
-			r, err := client.BandwidthAgreements(ctx, &rba)
-			if err != nil || r.GetStatus() == pb.AgreementsSummary_FAIL {
-				as.log.Warn("Agreementsender failed to send agreement to satellite : will retry", zap.Error(err))
-				continue
-			} else if r.GetStatus() == pb.AgreementsSummary_REJECTED {
-				//todo: something better than a delete here?
-				as.log.Error("Agreementsender had agreement explicitly rejected by satellite : will delete", zap.Error(err))
+			// CASE 1: successful connection with sat established but either failed or reject recd
+			if sat.GetStatus() != pb.AgreementsSummary_OK {
```

Comment:
what about a switch statement instead? That way, we don't have to call updateBandwidthAllocation on FAIL if default status is UNSENT anyway

Reply:
or just continue on FAIL